iam-login-service/src/main/java/it/infn/mw/iam/api/client/management/ClientManagementAPIController.java

```diff
@@ -157,6 +171,28 @@ public void disableClient(@PathVariable String clientId) {
     account.ifPresent(a -> managementService.updateClientStatus(clientId, false, a.getUuid()));
   }
 
+  @PatchMapping("/{clientId}/revoke-refresh-tokens")
+  @PreAuthorize("#iam.hasScope('iam:admin.write') or #iam.hasDashboardRole('ROLE_ADMIN')")
+  public void revokeRefreshTokens(@PathVariable String clientId) {
+    disableClient(clientId);
+    ClientDetailsEntity client = clientService.findClientByClientId(clientId)
+      .orElseThrow(ClientSuppliers.clientNotFound(clientId));
+    tokenService.getRefreshTokensForClient(client).forEach(tokenService::revokeRefreshToken);
+    rotateClientSecret(clientId);
```

Comment:
Same as above. There's another endpoint dedicated to this. I'd prefer to have a "revoke-refresh-token" and another endpoint called "resetClient" that does "disable-revoke-rotate-enable" flow. But this endpoint needs to return the new secret. We're moving in a direction that the secret is returned once it has been created and then hashed into the db.